src/plone/restapi/search/handler.py

```diff
@@ -74,16 +74,12 @@ def search(self, query=None):
         else:
             fullobjects = False
 
-        metadata_fields = query.pop("metadata_fields", [])
-        if not isinstance(metadata_fields, list):
```

Comment:
@gbastien question: is it safe to remove this isinstance check here? @sunew fixed an issue with this some time ago: https://github.com/plone/plone.restapi/commit/22d610e486aa4cd85ba87c85263cd3e68aea59cc

Since Sune added tests for this I guess it should be safe to remove. Though, maybe we need to double check if that use case is properly solved without this check...

Reply:
Ok. I should read the description before I guess. Found the metadata_field method in summary.py. Sorry for the noise and thank you for your PR!